src/taipy/core/submission/submission.py

```diff
@@ -44,19 +44,30 @@ class Submission(_Entity, _Labeled):
     def __init__(
         self,
         entity_id: str,
+        entity_type: str,
         id: Optional[str] = None,
         jobs: Optional[Union[List[Job], List[JobId]]] = None,
         creation_date: Optional[datetime] = None,
         submission_status: Optional[SubmissionStatus] = None,
         version: Optional[str] = None,
     ):
         self._entity_id = entity_id
+        self._entity_type = entity_type
         self.id = id or self.__new_id()
-        self._jobs: Union[List[Job], List[JobId], List] = jobs or []
+        self._job_ids = [job if isinstance(job, str) else job.id for job in jobs] if jobs else []
         self._creation_date = creation_date or datetime.now()
         self._submission_status = submission_status or SubmissionStatus.SUBMITTED
         self._version = version or _VersionManagerFactory._build_manager()._get_latest_version()
 
+        self.__abandoned = False
+        self.__canceled = False
+        self.__completed = False
+        self.__failed = False
```

Comment:
🤔 Those additional fields look redundant with there `_submission_status`...

Reply:
I have removed canceld and failed, but decided to keep abandoned and completed. I think they're still necessary, will look into if we can remove them also :D